components/results_table/commons/results_table.lua

```diff
@@ -5,3 +5,105 @@
 --
 -- Please see https://github.com/Liquipedia/Lua-Modules to contribute
 --
+
+local Class = require('Module:Class')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+
+---Note: This can be overwritten
+local Opponent = Lua.import('Module:Opponent', {requireDevIfEnabled = true})
+---Note: This can be overwritten
+local OpponentDisplay = Lua.import('Module:OpponentDisplay', {requireDevIfEnabled = true})
+
+local DEFAULT_VALUES = {
+	order = 'desc',
+	playerPrefix = 'p',
+	resolveOpponent = true,
+	playerLimit = 10,
+	coachLimit = 5,
+}
+local COACH_TYPE = 'coach'
+local TYPE_ALIASES = {
+	coach = COACH_TYPE,
+	['1v1'] = Opponent.solo,
+	player = Opponent.solo,
+	individual = Opponent.solo,
+	solo = Opponent.solo,
+	team = Opponent.team,
+}
+
+--- @class ResultsTable
+local ResultsTable = Class.new(function(self, ...) self:init(...) end)
+
+function ResultsTable:init(args)
+	self.args = args
+
+	self.pagename = mw.title.getCurrentTitle().text
+
+	if args.opponentLibrary then
+		Opponent = Lua.import('Module:'.. self.args.opponentLibrary, {requireDevIfEnabled = true})
+		self.opponentLibrary = Opponent
+	end
+	if args.opponentDisplayLibrary then
+		OpponentDisplay = Lua.import('Module:'.. self.args.opponentDisplayLibrary, {requireDevIfEnabled = true})
+	end
+
+	self.config = self:readConfig()
+
+	return self
+end
+
+function ResultsTable:readConfig()
+	local args = self.args
+
+	local config = {
+		order = args.order or DEFAULT_VALUES.order,
+		playerPrefix = args.prefixplayer or DEFAULT_VALUES.playerPrefix,
+		coachPrefix = args.prefixcoach or DEFAULT_VALUES.coachPrefix,
+		hideResult = Logic.readBool(args.hideresult),
+		resolveOpponent = Logic.readBool(args.resolvePlayer or args.resolveTeam or DEFAULT_VALUES.resolveOpponent),
+		gameIconsData = args.gameIcons,
+		opponent = mw.text.decode(args.coach or args.player or args.team or mw.title.getCurrentTitle().baseText),
+		opponentType = self:getOpponentType(),
+		onlyAchievements = Logic.readBool(args.achievements),
+	}
+
+	config.sort = args.sort or
+		(config.onlyAchievements and 'weight' or 'date')
+
+	config.limit = tonumber(args.limit) or
+		(config.onlyAchievements and 10 or 500)
```

Comment:
hmm
the commons and r6 version currently both set the 500 as default limit
there definitely are cases with >500
we could up it to 5000, higher will most likely break due to include size etc anyways
we also could add offset in case pages need to be broken up

Reply:
changed to 5k